Lib/compression/zstd/__init__.py

```diff
@@ -0,0 +1,293 @@
+"""Python bindings to Zstandard (zstd) compression library, the API style is
+similar to Python's bz2/lzma/zlib modules.
+"""
+
+__all__ = (
+    # From this file
+    "compressionLevel_values",
+    "get_frame_info",
+    "CParameter",
+    "DParameter",
+    "Strategy",
+    "finalize_dict",
+    "train_dict",
+    "zstd_support_multithread",
+    "compress",
+    "decompress",
+    # From _zstd
+    "ZstdCompressor",
+    "ZstdDecompressor",
+    "ZstdDict",
+    "ZstdError",
+    "get_frame_size",
+    "zstd_version",
+    "zstd_version_info",
+    # From zstd.zstdfile
+    "open",
+    "ZstdFile",
+)
+
+import enum
+import functools
+import dataclasses
+
+from compression.zstd.zstdfile import ZstdFile, open
+from _zstd import *
+
+import _zstd
+
+
+_ZSTD_CStreamSizes = _zstd._ZSTD_CStreamSizes
+_ZSTD_DStreamSizes = _zstd._ZSTD_DStreamSizes
+_train_dict = _zstd._train_dict
+_finalize_dict = _zstd._finalize_dict
+
+
+@dataclasses.dataclass(frozen=True)
+class _CompressionLevelValues:
+    default: int
+    min: int
+    max: int
```

Comment:
The only use of `dataclasses` is for two very simple immutable containers. For a slight boost to import performance, and to avoid implicitly exposing a ``dataclasses.is_dataclass()`` object, perhaps:

```suggestion
class _CLValues:
    __slots__ = 'default', 'min', 'max'

    def __init__(self, default, min, max):
        super().__setattr__('default', default)
        super().__setattr__('min', min)
        super().__setattr__('max', max)

    def __repr__(self):
        return (f'compression_level_values(default={self.default}, '
                f'min={self.min}, max={self.max})')

    def __setattr__(self, name, _):
        raise AttributeError(f"can't set attribute {name!r}")
```